node/config.ts

```diff
@@ -89,17 +98,26 @@ export const getFunctionConfig = async (
     log.user(stdout)
   }
 
-  try {
-    const collectorData = await fs.readFile(collector.path, 'utf8')
+  let collectorData: FunctionConfig = {}
 
-    return JSON.parse(collectorData) as FunctionConfig
+  try {
+    const collectorDataJSON = await fs.readFile(collector.path, 'utf8')
+    collectorData = JSON.parse(collectorDataJSON) as FunctionConfig
   } catch {
     handleConfigError(func, ConfigExitCode.UnhandledError, stderr, log, featureFlags)
-
-    return {}
   } finally {
     await collector.cleanup()
   }
+
+  if (!isValidOnError(collectorData.onError)) {
+    throw new BundleError(
```

Comment:
since this is new behaviour, I think it's fine to throw on error - no need to use the flag.